### ISSUE TITLE
Full support for dbt deps on Windows

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#778]
OSError: [WinError 17] The system cannot move the file to a different disk drive: 'C:\\Users\\UserName\\AppData\\Local\\Temp\\dbt-downloads\\75cce4837e0fd73e2934f79b9a5238a9' -> 'dbt_modules\\snowplow'

This can happen on Windows if the temp dir is on a different drive than the dbt project dir

**Comment by atharvai** [comment]
I spoke with Drew and suggested a workaround being able to override the temp dir. Of course this is no fix or solution but general good idea of having it configurable

**Comment by drewbanin** [comment]
Combining issue https://github.com/fishtown-analytics/dbt/issues/766

### dbt deps - symlink fails on windows (symbolic link privilege not held) 

Via slack:
```
2018-05-09 15:52:04,681: Tracking: do not track
2018-05-09 15:52:04,692: Installing ../
2018-05-09 15:52:04,693:   Creating symlink to local dependency.
2018-05-09 15:52:04,693: Encountered an error:
2018-05-09 15:52:04,693: symbolic link privilege not held
2018-05-09 15:52:04,695: Traceback (most recent call last):
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\main.py", line 41, in main
    results, succeeded = handle_and_check(args)
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\main.py", line 85, in handle_and_check
    task, res = run_from_args(parsed)
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\main.py", line 139, in run_from_args
    results = run_from_task(task, proj, parsed)
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\main.py", line 147, in run_from_task
    result = task.run()
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\task\deps.py", line 424, in run
    package.install(self.project)
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\task\deps.py", line 272, in install
    dbt.clients.system.make_symlink(src_path, dest_path)
  File "c:\users\bene\appdata\local\programs\python\python36-32\lib\site-packages\dbt\clients\system.py", line 106, in make_symlink
    return os.symlink(source, link_path)
OSError: symbolic link privilege not held
```

- Windows 10 Pro (developer mode enabled, can call `mklink` without elevated privileges)
- Using powershell
- dbt 10.0.0
- It works in an admin shell

We should either fix this, or add an option to not create a symlink

---

Via Ben Edwards in Slack:

> some other research that I have done, python's os.symlink will never work in windows unless run by an elevated user, even if the regular user can create symlinks (by enabling developer mode). The check in system.py only checks if the os supports symlink, which windows does, you just cant use it. A possible workaround I've seen is to detect windows, start a shell and call mklink from there.

**Comment by brian-bk** [comment]
Could just [set the TMPDIR env var](https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp)

**Comment by atharvai** [comment]
Here are some investigation scenarios that show that if the TMP dir and dbt project dir are on the OS drive the `dbt deps` is successful but not if they are on different drives https://gist.github.com/atharvai/822ef904c29bbd304bac07bb37233dab

**Comment by beckjake** [comment]
fixed in #1110

**Comment by chapmanjacobd** [comment]
I think this is the reason dbt has trouble installing deps on WSL2... is there a way to skip using the symlinks? when you only have 1 or 2 dbt projects it causes much much more headaches than the few MBs it saves
